pybossa/cache/site_stats.py

```diff
@@ -66,10 +84,16 @@ def n_anon_users():
 
 @memoize_with_l2_cache(timeout=ONE_DAY, timeout_l2=TWO_WEEKS,
                        key_prefix="site_n_tasks")
-def n_tasks_site():
+@allow_all_time_interval
+def n_tasks_site(interval='all'):
     """Return number of tasks in the server."""
-    sql = text('''SELECT COUNT(task.id) AS n_tasks FROM task''')
-    results = session.execute(sql)
+    sql = text('''
+        SELECT COUNT(task.id) AS n_tasks
+        FROM task WHERE
+        to_timestamp(task.created, 'YYYY-MM-DD"T"HH24:MI:SS.US') > current_timestamp - interval :interval;
```

Comment:
I know our existing code using the decorator to handle "days=all". I think the following SQL 1 would be much faster than SQL 2

SQL 1:
```sql
SELECT COUNT(task.id) AS n_tasks
        FROM task
``` 

SQL 2:
```sql
SELECT COUNT(task.id) AS n_tasks
        FROM task WHERE
        to_timestamp(task.created, 'YYYY-MM-DD"T"HH24:MI:SS.US') > current_timestamp - interval '999999 days';
``` 

I'm curious to research the result in QA.

Reply:
Ran some tests on QA:

SQL 1 (faster):
Planning Time: 0.501 ms
 Execution Time: 436.402 ms

SQL 2 (slower):
Planning Time: 0.070 ms
 Execution Time: 1893.185 ms

Note: SQL 2 takes around the same time as any other interval. (6 months, 12 months, 30 days, etc all run in the same time as 999999 days).